src/main/java/net/sf/jabref/gui/importer/fetcher/MedlineFetcher.java

```diff
@@ -130,109 +122,106 @@ public String getTitle() {
-                        frameOP.setStatus(Localization.lang("%0 import canceled", "Medline"));
-                        return false;
-                    }
+                // get the ids from entrez
+                SearchResult result = getIds(searchTerm, 0, 1);
 
-                    try {
-                        numberToFetch = Integer.parseInt(strCount.trim());
-                        break;
-                    } catch (NumberFormatException ex) {
-                        frameOP.showMessage(Localization.lang("Please enter a valid number"));
-                    }
+                if (result.count == 0) {
+                    status.showMessage(Localization.lang("No references found"),
+                            Localization.lang("Search %0", getTitle()), JOptionPane.INFORMATION_MESSAGE);
+                    return false;
                 }
-            }
 
-            for (int i = 0; i < numberToFetch; i += MedlineFetcher.PACING) {
-                if (!shouldContinue) {
-                    break;
+                int numberToFetch = result.count;
+                if (numberToFetch > MedlineFetcher.PACING) {
+                    boolean numberEntered = false;
+                    do {
+                        String strCount = JOptionPane.showInputDialog(Localization.lang("References found") + ": " + numberToFetch + " " +
+                                Localization.lang("Number of references to fetch?"), Integer.toString(numberToFetch));
+
+                        if (strCount == null) {
+                            status.setStatus(Localization.lang("%0 import canceled", getTitle()));
+                            return false;
+                        }
+
+                        try {
+                            numberToFetch = Integer.parseInt(strCount.trim());
+                            numberEntered = true;
+                        } catch (NumberFormatException ex) {
+                            status.showMessage(Localization.lang("Please enter a valid number"));
+                        }
+                    } while (!numberEntered);
                 }
 
-                int noToFetch = Math.min(MedlineFetcher.PACING, numberToFetch - i);
+                for (int i = 0; i < numberToFetch && shouldContinue; i += MedlineFetcher.PACING) {
+                    int noToFetch = Math.min(MedlineFetcher.PACING, numberToFetch - i);
 
-                // get the ids from entrez
-                result = getIds(searchTerm, i, noToFetch);
+                    // get the ids from entrez
+                    result = getIds(searchTerm, i, noToFetch);
 
-                List<BibEntry> bibs = fetchMedline(result.ids, frameOP);
-                for (BibEntry entry : bibs) {
-                    iIDialog.addEntry(entry);
+                    List<BibEntry> bibs = fetchMedline(result.ids, status);
+                    for (BibEntry entry : bibs) {
+                        dialog.addEntry(entry);
```

Comment:
I'm just curious, does the dialog object has a method which accepts a Collection/list of BibEntries?
(Not scope of this PR to add one, but would be nice to have such a method